src/dnx/pal.unix.cpp

```diff
@@ -47,33 +52,44 @@ BOOL GetFullPath(LPCTSTR szPath, LPTSTR szNormalizedPath)
     return TRUE;
 }
 
-HMODULE LoadNativeHost(LPCTSTR szHostModuleName)
+int CallApplicationMain(const char* moduleName, const char* functionName, CALL_APPLICATION_MAIN_DATA* data, TraceWriter traceWriter)
 {
-    LPTSTR localPath = GetNativeBootstrapperDirectory();
+    auto localPath =  GetNativeBootstrapperDirectory().append("/").append(moduleName);
 
-    strcat(localPath, "/");
-    strcat(localPath, szHostModuleName);
-
-    HMODULE hHost = dlopen(localPath, RTLD_NOW | RTLD_GLOBAL);
-
-    free(localPath);
-
-    return hHost;
-}
-
-BOOL FreeNativeHost(HMODULE hModule)
-{
-    if (hModule != nullptr)
+    void* host = nullptr;
+    try
     {
-        return dlclose(hModule);
+        host = dlopen(localPath.c_str(), RTLD_NOW | RTLD_GLOBAL);
+        if (!host)
+        {
+            throw std::runtime_error(std::string("Failed to load: ").append(moduleName));
+        }
+
+        traceWriter.Write(std::string("Loaded module: ").append(moduleName), true);
```

Comment:
I remember fixing this - must have overwritten the file when moving changes between boxes. Should be `std::string` everywhere since this is a platform specific file and we know what types we should be using.